src/test/java/org/opentripplanner/middleware/utils/ItineraryExistenceCheckerTest.java

```diff
@@ -0,0 +1,79 @@
+package org.opentripplanner.middleware.utils;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.otp.OtpDispatcher;
+import org.opentripplanner.middleware.otp.OtpDispatcherResponse;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+
+import static org.opentripplanner.middleware.TestUtils.TEST_RESOURCE_PATH;
+
+/**
+ * Tests for checking the existence of trips from query strings.
+ */
+public class ItineraryExistenceCheckerTest {
+    /**
+     * Map query strings to mock OTP responses.
+     * We use {@link HashMap#get} function to mock {@link OtpDispatcher#sendOtpPlanRequest}
+     * (both have the same signature).
+     */
+    private static HashMap<String, OtpDispatcherResponse> queryToResponse;
+
+    /** Mock (significantly abbreviated) OTP response when an itinerary is not found. */
+    private final static String MOCK_RESPONSE_WITH_ERROR = "{\"requestParameters\":{},\"error\":{\"id\":404,\"msg\":\"No trip found...\",\"message\":\"PATH_NOT_FOUND\",\"noPath\":true},\"debugOutput\":{}}";
```

Comment:
Perhaps change to JSON ObjectNodes?

Reply:
Or import a file from the resources folder.

Reply:
Good idea! I updated the class to use the error response file that is already in the resources folder. See https://github.com/ibi-group/otp-middleware/pull/77/commits/d331144c173deeb3e2e5cc8542f174d42312ff80.